core/templates/dev/head/pages/library/library.html

```diff
@@ -90,20 +90,16 @@ <h2 class="oppia-library-h2" translate="I18N_LIBRARY_SUB_HEADER"></h2>
                translate="I18N_LIBRARY_NO_EXPLORATION_GROUPS">
           </div>
 
-          <div class="oppia-library-group" ng-repeat="group in libraryGroups track by $index">
+          <div class="oppia-library-group" ng-repeat="group in libraryGroups track by $index" ng-mouseenter="setActiveGroup($index)" ng-mouseleave="clearActiveGroup()">
             <div class="row oppia-library-group-header-container">
-              <h2 class="oppia-library-group-header">
-                <a ng-if="group.has_full_results_page"
-                   ng-click="showFullResultsPage(group.categories, group.full_results_url)"
-                   translate="<[group.header_i18n_id]>">
-                </a>
+              <h2 ng-class="{'oppia-library-group-header-active': activeGroupIndex === $index, 'oppia-library-group-header': activeGroupIndex != $index }">
+                <span ng-if="group.has_full_results_page" translate="<[group.header_i18n_id]>"
```

Comment:
No, it should still look like the mocks. But you can style using CSS to remove the underline (see the text-decoration attr).